test/shared/LiquidityOperations.sol

```diff
@@ -93,6 +93,31 @@ abstract contract LiquidityOperations is CommonBase {
         return abi.decode(result[0], (BalanceDelta));
     }
 
+    // TODO: organize somewhere else, or rename this file to NFTLiquidityHelpers?
+    function permit(address signer, uint256 privateKey, uint256 tokenId, address operator, uint256 nonce) internal {
+        bytes32 digest = getDigest(operator, tokenId, 1, block.timestamp + 1);
+
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(privateKey, digest);
+
+        vm.prank(signer);
```

Comment:
imo the whole point in a permit is that the signer often isnt the person who sends it. I think you should have a parameter of this function `caller` and do `vm.prank(caller)`. Then some tests you can pass in `signer` and others you can explicitly test that someone else can pass it in

Reply:
e.g. a test where the `caller` is the `operator`!

Reply:
ok i see youre already testing that - just not in this helper function. Thats fair. I _think_ I'd still rather this was `operator` not `signer` pranking just because the `signer` wouldnt do this - they'd just call `approve` instead